lib/topology/platforms/shell.py

```diff
@@ -585,29 +599,54 @@ def connect(self, connection=None):
         )
 
         self._connections[connection] = spawn
+        index = None
 
         try:
+            # Setup shell before using it
+            self._pre_setup_shell(connection)
+
             # If connection is via user
             if self._user is not None:
-                spawn.expect(
-                    [self._user_match], timeout=self._timeout
+                index = spawn.expect(
```

Comment:
This looks like functionality that could be useful, can you please provide more details regarding the connection process you want Topology to follow?